service/src/test/java/io/camunda/service/license/CamundaLicenseTest.java

```diff
@@ -7,27 +7,38 @@
  */
 package io.camunda.service.license;
 
+import static org.junit.jupiter.api.Assertions.assertFalse;
 import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
 
+import org.camunda.bpm.licensecheck.InvalidLicenseException;
+import org.camunda.bpm.licensecheck.LicenseKey;
 import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
 
 public class CamundaLicenseTest {
-  public static final String VALID_LICENSE = "valid license";
 
   @Test
-  public void shouldReturnTrueWhenLicenseIsValid() {
-    // given
-    final CamundaLicense camundaLicense =
-        new CamundaLicense(new MockEnvironmentVariableReaderWithValidLicense());
+  public void shouldReturnTrueWhenLicenseIsValid() throws InvalidLicenseException {
+    final CamundaLicense testLicense = spy(CamundaLicense.class);
+    final LicenseKey mockKey = mock(LicenseKey.class);
 
-    assertTrue(camundaLicense.isValid());
+    doReturn(mockKey).when(testLicense).getLicenseKey(anyString());
```

Comment:
💭 🔧 Wondering whether it makes sense to test the protected methods here. I feel like this is too much implementation detail, we should rather focus on the API (public methods) and behavior of classes. 

Meaning, that if I create a CamundaLicense with a specific key and I call it valid I expect it is either valid or not (depending on the input). If I can't change the behavior of the class and test it appropriately, it might based on the class design. So we could improve the design to be able to inject dependencies, like a LicenseKey factory or something.

Just me 2ct to this.

Reply:
Why does this matter? The test is now coupled to the structure of the class. If I change any implementation detail of the class, there is a high chance that the tests are broken, even if the behavior of the class didn't even change.

Reply:
Makes sense! I refactored the tests to work with the public methods. What's mocked is just the call to get the license

I ran out of time for the day to finish the other two tests, but I'll get those written up tomorrow

Reply:
Tests pushed!